reference/app-availability.html.md.erb

```diff
@@ -0,0 +1,124 @@
+---
+title: App Availability and Resiliency
+layout: docs
+sitemap: false
+nav: firecracker
+---
+
+A resilient app is resistant to events like hardware failures or outages. Redundancy, supported by monitoring and scaling, is one of the ways to make your app more resilient, and we provide some built-in features to make this easier.
+
+But first, it's important to understand what your app needs to take care of.
+
+## What your app needs to do
+
+<div class="callout">
+<p><b>A note about Fly Proxy</b>: 
+Fly Proxy does a lot of work in the background, applying handlers, managing traffic and load balancing, and managing connections to services. Fly Proxy also automatically starts and stops Machines based on traffic to your app. The proxy works for [process groups with services defined](/docs/getting-started/app-services/#services-routed-with-fly-proxy).
+</div>
+
+Since Fly Proxy does load balancing by default, you need to consider how your app handles:
+
+* session storage, so that users stay logged in when their session moves to a different server
+* persistent file storage and replication, as discussed in the following paragraphs
+
+For databases, you'll need to follow the recommendations specific to your deployment, including how to set up clusters for replication and failover.
+
+If you're using Fly Postgres, our un-managed database, then you need to [manage backups, recovery, replication, and scaling](/docs/postgres/getting-started/what-you-should-know/) yourself. For information about high availability with Fly Postgres, refer to [High Availability & Global Replication](/docs/postgres/advanced-guides/high-availability-and-global-replication/).
+
+When you use [Fly volumes](/docs/reference/volumes/) with your app, your app needs to implement replication and backups. Volumes are block devices that live on a single disk array and get mounted directly by your Machine. There's no magic. And because there's no built-in replication between volumes, Fly.io never automatically starts multiple Machines with volumes on the first deploy. If your app provides a clustering mechanism with data replication, like most databases do, we recommend you take advantage of that and use multiple volumes per app. We place your app's volumes in different hardware zones within a region to mitigate hardware failures.
```

Comment:
"Fly Volumes"

Reply:
> And because there's no built-in replication between volumes, Fly.io never automatically starts multiple Machines with volumes on the first deploy.

I'd leave this point until after the feature is introduced! `fly deploy` is the only thing that'll automatically create an extra Machine and it's super confusing outside that context.

Reply:
> If your app provides a clustering mechanism with data replication, like most databases do, we recommend you take advantage of that and use multiple volumes per app. 

What about "we recommend that you take advantage of that and run multiple instances (each with its own attached volume)"? You can't run a volume without a VM, but people don't always get that off the bat.

Reply:
> We place your app's volumes in different hardware zones within a region to mitigate hardware failures.

"When possible, we place your app's volumes in different hardware zones within a region to mitigate the impact of potential hardware failures." ?